sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -271,7 +271,15 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         ignoreIfExists)
     } else {
       val tableWithDataSourceProps = tableDefinition.copy(
-        schema = hiveCompatibleSchema,
+        schema = hiveCompatibleSchema match {
```

Comment:
nit: can we rename the variable name here, it seems a bit weird that we are finding incompatible types from a compatible schema